### PR TITLE
BEI-20: Create the users table

### PR DESCRIPTION
**Pull request by hizikex**
#### WHY THIS PR?
We need a model for users to be uniquely identified and authenticated.

#### WHAT DOES THIS PR DO?
It creates a model for users to be individually identified and be authenticated.
It creates a secured and organized approach to managing user-related data.

#### HOW CAN THIS PR BE TESTED?
-When a datatype does not correspond to the given datatype of a field.

#### THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: src/core/database/sequelize.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const sequelizeConfig: Options = require('../config/config')[process.env.NODE_ENV || 'development'];
+
+const { database, username, password, ...otherConfig } = sequelizeConfig;
+const sequelize = new Sequelize(database || '', username || '', password, otherConfig);
+
+// sequelize.authenticate().then(() => {
+//   logger.info('Database connection successful');
+// });
```
Comment (RICHYDOS) [inline src/core/database/sequelize.ts:11]: If commented code is not needed delete it.

File: src/core/interfaces/user.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export interface UserAttributes {
+    id: number;
+    username: string;
+    email: string;
+    password: string;
+    bio: string;
+    image: string;
```
Comment (RICHYDOS) [inline src/core/interfaces/user.ts:7]: Make these two optional to conform with the models and migration files

File: src/core/models/users.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import { sequelize } from '../database/sequelize';
+import { UserAttributes } from '../interfaces/user';
+
+export class User extends Model<InferAttributes<User>, InferCreationAttributes<User, { omit: 'id' }>> implements UserAttributes {
+  declare id: CreationOptional<number>;
+  declare username: string;
+  declare email: string;
+  declare password: string;
+  declare bio: string;
+  declare image: string;
```
Comment (RICHYDOS) [inline src/core/models/users.ts:11]: Same thing here regarding optional when creating a user

**Review by RICHYDOS — CHANGES_REQUESTED** [review]
Well Done!

File: src/core/database/sequelize.ts (hunk 1 above)
Comment (hizikex) [reply src/core/database/sequelize.ts:11]: Thanks

**Review by RICHYDOS — APPROVED** [review]

**Review by LucianaDera — APPROVED** [review]

**Review by mushkaa — APPROVED** [review]

**Pull request merged**